autoload/ale/engine.vim

```diff
@@ -552,6 +553,7 @@ function! s:RunJob(options) abort
         \   'buffer': l:buffer,
         \   'output': [],
         \   'next_chain_index': l:next_chain_index,
+        \   'wd': getcwd(),
```

Comment:
This is also possibly the wrong working directory, as this could run from some buffer other than the one you are checking.